### PR TITLE
Fix e2e + unit tests - also update dependencies

### PR DESCRIPTION
**Pull request by mzedel**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -67,6 +74,14 @@ module.exports = {
       }
     ]
   },
+  optimization: {
+    minimize: process.env.NODE_ENV === 'production',
```
Comment (merlin-northern) [inline webpack.config.js:78]: so it does not influence development builds?

File: tests/e2e_tests/cypress/integration/04-deviceDetails.spec.js [hunk 2]
```diff
@@ -29,11 +29,9 @@ context('Layout assertions', () => {
         () =>
           cy
             .get('.deviceListItem')
-            .click()
             // the deviceconnect connection might not be established right away,
             // due to polling intervals the fastest way to refresh is collapsing & expanding again...
```
Comment (merlin-northern) [inline tests/e2e_tests/cypress/integration/04-deviceDetails.spec.js:33]: does this comment still make sense?

File: tests/setupTests.js [hunk 3]
```diff
@@ -21,6 +21,7 @@ let server;
 const oldWindowLocation = window.location;
 const oldWindowSessionStorage = window.sessionStorage;
 
+jest.retryTimes(3);
```
Comment (merlin-northern) [inline tests/setupTests.js:24]: nitpick: a const for the 3?

**Review by merlin-northern — APPROVED** [review]
lgtm! take or leave the comments.

File: tests/setupTests.js (hunk 3 above)
Comment (mzedel) [reply tests/setupTests.js:24]: That whole file is basically just a collection of consts 😅 but it's probably as good as opportunity as any to clean that up a bit

File: tests/e2e_tests/cypress/integration/04-deviceDetails.spec.js (hunk 2 above)
Comment (mzedel) [reply tests/e2e_tests/cypress/integration/04-deviceDetails.spec.js:33]: Yep, the clicking will still be done repeatedly, but now just once and then wait until the animation had settled, Check if the button is usable and repeat

File: webpack.config.js (hunk 1 above)
Comment (mzedel) [reply webpack.config.js:78]: Yep, the dev builds are faster and the prod builds result in a smaller payload this way...

Comment (mzedel) [reply webpack.config.js:78]: Previously this happened automatically, but due to the improved loader we're now using, that setting is required

**Review by merlin-northern — APPROVED** [review]
thank you for the answers!

**Pull request merged**